### PR TITLE
`flex-await` improvements

### PR DESCRIPTION
**Pull request by ashtum**

### CODE REVIEW

File: example/client/flex_await/Jamfile [hunk 1]
```diff
@@ -8,11 +8,13 @@
 #
 
 using openssl ;
+import ac ;
 
 project
     : requirements
       $(c11-requires)
       <library>/boost/http_proto//boost_http_proto
+      [ ac.check-library /boost/http_proto//boost_http_proto_zlib : <library>/boost/http_proto//boost_http_proto_zlib : ]
```
Comment (vinniefalco) [inline example/client/flex_await/Jamfile:17]: Don't you need to also conditionally compile the call to install the service?

Comment (ashtum) [reply example/client/flex_await/Jamfile:17]: I used `BOOST_HTTP_PROTO_HAS_ZLIB` for that:
https://github.com/cppalliance/http_io/blob/24f1d92398e372bb5ed350ced7a668c03dafd8a1/example/client/flex_await/main.cpp#L334-L338

Comment (vinniefalco) [reply example/client/flex_await/Jamfile:17]: Who sets `BOOST_HTTP_PROTO_HAS_ZLIB ` ?

Comment (ashtum) [reply example/client/flex_await/Jamfile:17]: Build system. Yeah it doesn't look a good idea for an example.

**Pull request merged**